operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java

```diff
@@ -139,13 +174,14 @@ public static Step createAvailableStep(String reason, Step next) {
   }
 
   /**
-   * Asynchronous step to set Domain condition to Failed after an asynchronous call failure.
+   * Asynchronous step to set Domain condition to Failed after an asynchronous call failure and
+   * and to generate DOMAIN_PROCESSING_FAILED event.
    *
    * @param callResponse the response from an unrecoverable call
    * @param next Next step
    * @return Step
    */
-  public static Step createFailedStep(CallResponse<?> callResponse, Step next) {
+  public static Step createFailedAndEventStep(CallResponse<?> callResponse, Step next) {
```

Comment:
The name is a bit confusing as "Failed" is an adjective and "EventStep" is a modified noun. The two are not parallel and should not be connected with "and." I cannot tell the intent. Is it creating a step to report a failure event? If so, perhaps `createFailureEventStep` would be a better name.

Or is it necessarily creating two steps? If so, "Step" should be plural, assuming that they are closely related. 

In general, "and" in a method name is a warning that the method is doing two things, not just one.

Reply:
I changed the name to createFailureRelatedSteps.